### ISSUE TITLE
materialized view not exist  when refreshing and drop task error

### ISSUE THREAD

**Issue by medivh666** [StarRocks/starrocks#38188]
I found /opt/fe/log/fe.warn.log 

```
"materialized view:24777583 in database:mx_data do not exist when refreshing"
```
But materialized view ID=24777583  exit in information_schema.tasks ;
I try drop it ,but report error.

```
MySQL [mx_data]> drop task mv-24777583;
ERROR 1064 (HY000): Getting syntax error at line 1, column 12. Detail message: Unexpected input '-', the most similar input is {<EOF>, ';'}.
```

```
2023-12-31 22:21:52,231 WARN (pool-23-thread-3|263) [TaskRun.refreshTaskProperties():137] materialized view:24777583 in database:mx_data do not exist when refreshing
2023-12-31 22:21:52,248 WARN (pool-23-thread-3|263) [PartitionBasedMvRefreshProcessor.prepare():631] materialized view:24777583 in database:mx_data do not exist when refreshing
2023-12-31 22:21:52,248 WARN (pool-23-thread-3|263) [TaskRunExecutor.lambda$executeTaskRun$0():54] failed to execute TaskRun.
com.starrocks.sql.common.DmlException: database mx_data do not exist.
        at com.starrocks.scheduler.PartitionBasedMvRefreshProcessor.prepare(PartitionBasedMvRefreshProcessor.java:633) ~[starrocks-fe.jar:?]
        at com.starrocks.scheduler.PartitionBasedMvRefreshProcessor.processTaskRun(PartitionBasedMvRefreshProcessor.java:177) ~[starrocks-fe.jar:?]
        at com.starrocks.scheduler.TaskRun.executeTaskRun(TaskRun.java:189) ~[starrocks-fe.jar:?]
        at com.starrocks.scheduler.TaskRunExecutor.lambda$executeTaskRun$0(TaskRunExecutor.java:47) ~[starrocks-fe.jar:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1604) ~[?:1.8.0_382]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_382]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_382]
        at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_382]
```

### StarRocks version (Required)
Version:3.1.4
Shared-data deployment

**Comment by LiShuMing** [comment]
please try `drop task `mv-24777583``, you should add ``` as a specific identity for task name?

**Comment by medivh666** [comment]
MySQL [mx_data]> drop task `mv-24777583`;
ERROR 1064 (HY000): Unexpected exception: Can not drop task generated by materialized view. You can use DROP MATERIALIZED VIEW to drop task, when the materialized view is deleted, the related task will be deleted automatically.

**Comment by mofeiatwork** [comment]
fixed by #39932
